test/unit/sdam/monitor.test.ts

```diff
@@ -579,4 +582,32 @@ describe('monitoring', function () {
       });
     });
   });
+
+  describe('Heartbeat duration', function () {
+    let client: MongoClient;
+    let serverHeartbeatFailed;
+
+    beforeEach(async function () {
+      // Artificially make creating a connection take 200ms
+      sinon.stub(net, 'createConnection').callsFake(function () {
+        const socket = new net.Socket();
+        setTimeout(() => socket.emit('connect'), 80);
+        socket.on('data', () => socket.destroy(new Error('I am not real!')));
+        return socket;
+      });
+
+      client = new MongoClient(`mongodb://localhost:1`, { serverSelectionTimeoutMS: 200 });
+      client.on('serverHeartbeatFailed', ev => (serverHeartbeatFailed = ev));
+    });
+
+    afterEach(function () {
+      sinon.restore();
```

Comment:
Do we need to destroy the socket in afterEach? In case something went wrong and we haven't destroyed it on data callback.